internal/build/resolver/options/options.go

```diff
@@ -0,0 +1,75 @@
+package options
+
+import (
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
+	"github.com/go-git/go-git/v5"
+)
+
+// OptionsFn is a function to apply modifications to the list builds API request ie. for adding additional filters
+type OptionsFn func(*buildkite.BuildsListOptions) error
+
+type AggregateResolver []OptionsFn
+
+func (ar AggregateResolver) WithResolverWhen(condition bool, resovler OptionsFn) AggregateResolver {
+	if condition {
+		return append(ar, resovler)
+	}
+	return ar
+}
+
+// ResolveBranchFromFlag returns a function that is used to add a branch filter to a build list options
+func ResolveBranchFromFlag(branch string) OptionsFn {
+	return func(options *buildkite.BuildsListOptions) error {
+		if branch != "" && len(options.Branch) == 0 {
+			options.Branch = append(options.Branch, branch)
+		}
+		return nil
+	}
+}
+
+// ResolveBranchFromRepository returns a function that is used to add a branch filter to a build list options
+func ResolveBranchFromRepository(repo *git.Repository) OptionsFn {
+	return func(options *buildkite.BuildsListOptions) error {
+		var branch string
+		if repo != nil && len(options.Branch) == 0 {
+			head, err := repo.Head()
+			if err != nil {
+				return err
+			}
```

Comment:
Thinking about the case where `bk` is run outside a repository, this would error and not be able to run the command. But perhaps it should just silently not add a branch filter instead. That way its still usable outside a repo if no explicit `--branch` flag is used